llm/llama.go

```diff
@@ -218,6 +224,72 @@ type llama struct {
 	Running
 }
 
+var errNoGPU = errors.New("nvidia-smi command failed")
+
+// CheckVRAM returns the available VRAM in MiB on Linux machines with NVIDIA GPUs
+func CheckVRAM() (int, error) {
+	cmd := exec.Command("nvidia-smi", "--query-gpu=memory.free", "--format=csv,noheader,nounits")
```

Comment:
Would it be better to use `memory.total`? I believe worst case nvidia GPUs will begin to [evict](https://developer.nvidia.com/blog/improving-gpu-memory-oversubscription-performance/) existing memory pages to system memory (RAM) which, in theory, should still be fast.